lib/chef/resource/rhsm_register.rb

```diff
@@ -121,24 +121,30 @@ class RhsmRegister < Chef::Resource
       end
 
       action_class do
+        #
+        # @return [Symbol] dnf_package or yum_package depending on OS release
+        #
         def package_resource
           node["platform_version"].to_i >= 8 ? :dnf_package : :yum_package
         end
 
+        #
+        # @return [Boolean] is katello-ca-consumer installed
+        #
         def registered_with_rhsm?
-          # FIXME: use shell_out
-          cmd = Mixlib::ShellOut.new("subscription-manager status", env: { LANG: "en_US" })
-          cmd.run_command
-          !cmd.stdout.match(/Overall Status: Unknown/)
+          @registered ||= !shell_out("subscription-manager status").stdout.include?("Overall Status: Unknown")
```

Comment:
Was leaving off the LANG intentional?

Reply:
shell_out should be defaulting to en-us with the new changes